addons/binding/org.openhab.binding.fritzaha/src/main/java/org/openhab/binding/fritzaha/internal/discovery/FritzBoxDiscoveryService.java

```diff
@@ -0,0 +1,110 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.fritzaha.internal.discovery;
+
+import static org.openhab.binding.fritzaha.FritzAhaBindingConstants.*;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.openhab.binding.fritzaha.handler.FritzAhaBridgeHandler;
+import org.openhab.binding.fritzaha.handler.FritzAhaHandler;
+import org.openhab.binding.fritzaha.internal.model.DeviceModel;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Discover all AHA (AVM Home Automation) devices connected to a FRITZ!Box
+ * device.
+ *
+ * @author Robert Bausdorf
+ * @since 2.0
+ */
+public class FritzBoxDiscoveryService extends AbstractDiscoveryService {
```

Comment:
you might also want to override the methods ```startBackgroundDiscovery()``` and ```stopBackgroundDiscovery()``` - in these you could schedule a regular job that does an automatic scan in the background - every few minutes will be good enough for sure.

One feature I would actually expect is that as soon as the Fritz!Box is successfully accessed for the first time after the handler initialization, it should trigger a scan for things on it. At the moment you have to manually trigger a scan after having configured the Fritz!Box as a bridge.